lib/improver/tests/utilities/test_solar.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the solar calculations in solar.py """
+
+import unittest
+import numpy as np
+
+from iris.tests import IrisTest
+
+from improver.utilities.solar import (
+    solar_declination, solar_hour_angle, solar_elevation,
+    daynight_terminator)
+
+
+class Test_solar_declination(IrisTest):
+    """Test Solar declination."""
+
+    def test_basic_solar_declination(self):
+        """Test the calc of declination for different days of the year"""
+        day_of_year = [1, 10, 100, 365]
+        expected_result = [-23.1223537018, -22.1987731152,
+                           7.20726681123, -23.2078460336]
+        for i, dayval in enumerate(day_of_year):
+            result = solar_declination(dayval)
+            self.assertAlmostEqual(result, expected_result[i])
```

Comment:
Test what happens if invalid number supplied (e.g. 0 or 367). Perhaps add a exception to the function for these.